### PR TITLE
Enable caching for requests from data studio

### PR DESCRIPTION
**Pull request by licitdev**
Enable caching for requests from data studio when `CACHE_AUTO_PURGE` is enabled.

Collections added to `CACHE_AUTO_PURGE_IGNORE_LIST` will not cause the cache to auto purge.

`directus_activity` and `directus_presets` are defaults as there is no need for cache to purge when using the data studio (Eg: Updating preset when browsing).

Data studio have access to "fresh" data, bypassing the cache accordingly.

| CACHE_AUTO_PURGE | CACHE_AUTO_PURGE_IGNORE_LIST | Request URL              | Will Cache? | Data Studio Skips Cache? |
|------------------|------------------------------|--------------------------|-------------|--------------------------|
| false            | -                            | /items/test_collection   | YES         | YES                      |
| true             | ignore_collection            | /items/test_collection   | YES         | NO                       |
| true             | ignore_collection            | /items/ignore_collection | YES         | YES                      |

Closes ENG-940.

### CODE REVIEW

File: api/src/services/revisions.ts [hunk 1]
```diff
@@ -22,4 +22,36 @@ export class RevisionsService extends ItemsService {
...
+	override async createMany(data: Partial<Item>[], opts?: MutationOptions): Promise<PrimaryKey[]> {
+		return super.createMany(data, this.setDefaultOptions(opts));
+	}
+
+	override async updateOne(key: PrimaryKey, data: Partial<Item>, opts?: MutationOptions): Promise<PrimaryKey> {
+		return super.updateOne(key, data, this.setDefaultOptions(opts));
+	}
+
+	override async updateMany(keys: PrimaryKey[], data: Partial<Item>, opts?: MutationOptions): Promise<PrimaryKey[]> {
+		return await super.updateMany(keys, data, this.setDefaultOptions(opts));
```
Comment (azrikahar) [inline api/src/services/revisions.ts:55]:
```suggestion
		return super.updateMany(keys, data, this.setDefaultOptions(opts));
```

I presume we can remove this, but I might be missing some context 🧐

Comment (br41nslug) [reply api/src/services/revisions.ts:55]: I dont think it technically matters as promises are collapsed when awaiting. However i think i prefer the explicit `await` here 🤔

Comment (licitdev) [reply api/src/services/revisions.ts:55]: Good catch! 👍 Not sure how the await gotten there but let's remove the await as we are not doing any further processing with the result in this case.

File: api/src/utils/should-skip-cache.ts [hunk 2]
```diff
@@ -19,13 +21,29 @@ export function shouldSkipCache(req: Request): boolean {
...
+
+	function checkAutoPurge() {
+		if (env['CACHE_AUTO_PURGE'] === false) return true;
+
+		const path = url.parse(req.originalUrl).pathname;
+
+		for (const collection of env['CACHE_AUTO_PURGE_IGNORE_LIST']) {
+			const ignoredPath = getEndpoint(collection);
+
+			if (path?.startsWith(ignoredPath)) {
```
[the remark this answers is not in the document]
Comment (licitdev) [reply api/src/utils/should-skip-cache.ts:42]: Yes! Thank you! 👍

File: api/src/services/items.ts [hunk 3]
```diff
@@ -298,7 +298,12 @@ export class ItemsService<Item extends AnyItem = AnyItem> implements AbstractSer
 			}
 		}
 
-		if (this.cache && env['CACHE_AUTO_PURGE'] && opts.autoPurgeCache !== false) {
+		if (
+			this.cache &&
+			env['CACHE_AUTO_PURGE'] &&
+			opts.autoPurgeCache !== false &&
+			!env['CACHE_AUTO_PURGE_IGNORE_LIST'].includes(this.collection)
```
[the remark this answers is not in the document]
Comment (licitdev) [reply api/src/services/items.ts]: Great idea! Added a `shouldClearCache` util and applied the changes with all existing checks. Thanks!

**Review by br41nslug — APPROVED** [review]
Looking good. However perhaps an extra note in the docs on how the `PUBLIC_URL` figures into this would be nice (for example the different behavior between relative and absolute URLs)

**Review by paescuj — APPROVED** [review]
LGTM 🥇

**Pull request merged**